nix/pkgs/haskell/haskell.nix

```diff
@@ -59,9 +58,138 @@ let
+      package plutus-metatheory
+        tests: False
+
+      package prettyprinter-configurable
+        tests: False
+
+      package small-steps
+        tests: False
+
+      package small-steps-test
+        tests: False
+
+      package byron-spec-chain
+        tests: False
+
+      package cardano-node
+        flags: -systemd
+
+      package cardano-config
+        flags: -systemd
+    '';
     modules = [
-      {
-        reinstallableLibGhc = false;
+      # Allow reinstallation of Win32
+      ({ pkgs, ... }: lib.mkIf pkgs.stdenv.hostPlatform.isWindows {
+        nonReinstallablePkgs =
+          [
+            "rts"
+            "ghc-heap"
+            "ghc-prim"
+            "integer-gmp"
+            "integer-simple"
+            "base"
+            "deepseq"
+            "array"
+            "ghc-boot-th"
+            "pretty"
+            "template-haskell"
+            # ghcjs custom packages
+            "ghcjs-prim"
+            "ghcjs-th"
+            "ghc"
+            "array"
+            "binary"
+            "bytestring"
+            "containers"
+            "filepath"
+            "ghc-compact"
+            "ghc-prim"
+            # "ghci" "haskeline"
+            "hpc"
+            "mtl"
+            "parsec"
+            "text"
+            "transformers"
+            "xhtml"
+            # "stm" "terminfo"
+          ];
+        packages.Win32.components.library.build-tools = lib.mkForce [ ];
+      })
+      ({ pkgs, ... }: lib.mkIf (pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform) {
+        packages = {
+          # Things that need plutus-tx-plugins
+          marlowe.package.buildable = false; # Would also require libpq
+          marlowe-actus.package.buildable = false;
+          marlowe-dashboard-server.package.buildable = false;
+          marlowe-playground-server.package.buildable = false; # Would also require libpq
+          marlowe-symbolic.package.buildable = false;
+          playground-common.package.buildable = false;
+          plutus-benchmark.package.buildable = false;
+          plutus-chain-index.package.buildable = false;
+          plutus-contract.package.buildable = false;
+          plutus-errors.package.buildable = false;
+          plutus-ledger.package.buildable = false;
+          plutus-pab.package.buildable = false;
+          plutus-playground-server.package.buildable = false; # Would also require libpq
+          plutus-use-cases.package.buildable = false;
+          web-ghc.package.buildable = false;
+          # Needs agda
+          plutus-metatheory.package.buildable = false;
+          # These need R
+          plutus-core.components.benchmarks.cost-model-test.buildable = lib.mkForce false;
+          plutus-core.components.benchmarks.update-cost-model.buildable = lib.mkForce false;
+          # Windows build of libpq is marked as broken
+          fake-pab.package.buildable = false;
+        };
+      })
+      ({ pkgs, ... }:
+        let
+          symlinkDlls = ''
```

Comment:
needs a comment. Where does this come from? What is it for? Can we fix this in haskell.nix?